rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/NodeSourcePolicyFactory.java

```diff
@@ -52,16 +51,26 @@
     private static Collection<Class<?>> supportedPolicies;
 
     /**
-     * Creates a new node source policy using reflection mechanism.
+     * Creates a new active node source policy using reflection mechanism.
      *
      * @param policyClassName full class name of the policy
      * @param infrastructureType infrastructure class name (for compatibility check)
      * @param policyParameters policy parameters
-     * @return new instance of the node source policy
+     * @return a stub of the node source policy
      */
-    public static NodeSourcePolicy create(String policyClassName, String infrastructureType,
+    public static NodeSourcePolicy createAndActivate(String policyClassName, String infrastructureType,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "infrastructureType". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)